### PR TITLE
fix: Restrict which files Vite publishes

### LINKED ISSUE #12362: [Vite] restrict which folders are served
### Description of the bug / feature
By default, Vite publishes the whole folder where it is run. This works fine for a frontend only project but you would not want to publish all your Maven/Gradle projects on the local network.

The following folders should be ok to publish
- `frontend`
- `node_modules`
- `target/flow-frontend`
- `target/frontend`

### PR DESCRIPTION
**Pull request by Artur-**
Fixes #12362

### CODE REVIEW

**Review by caalador — APPROVED** [review]

**Comment by caalador** [comment]
The access tests never see that the server would be up and running.
```
13:13:21   ------------------ Starting Frontend compilation. ------------------
13:13:22   
13:13:22   ----------------- Frontend compiled successfully. -----------------
13:13:22   
13:13:29   [INFO] Running com.vaadin.viteapp.BasicsIT
13:13:29   [INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 14.113 s - in com.vaadin.viteapp.BasicsIT
13:13:29   [INFO] Running com.vaadin.viteapp.FileAccessIT
13:13:29   [ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 5.222 s <<< FAILURE! - in com.vaadin.viteapp.FileAccessIT
13:13:29   [ERROR] mostFoldersNotAccessible(com.vaadin.viteapp.FileAccessIT)  Time elapsed: 5.222 s  <<< ERROR!
13:13:29   java.lang.IllegalStateException: Dev server never finished

```

**Comment by Artur-** [comment]
Increased the test timeout...

**Review by joheriks — APPROVED** [review]

File: flow-tests/test-frontend/vite-basics/src/test/java/com/vaadin/viteapp/FileAccessIT.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                    e.getMessage().contains(
+                            "Server returned HTTP response code: 403"));
+        }
+
+    }
+
+    private URL getFsUrl(String fileInProject) throws IOException {
+        String currentPath = new java.io.File(".").getCanonicalPath();
+        return new URL("http://localhost:8888/VAADIN/@fs" + currentPath + "/"
+                + fileInProject);
```
Comment (caalador) [inline flow-tests/test-frontend/vite-basics/src/test/java/com/vaadin/viteapp/FileAccessIT.java:75]: `java.lang.AssertionError: Request for http://localhost:8888/VAADIN/@fsC:\Users\mikae\Code\flow\flow-tests\test-frontend\vite-basics/target/vaadin-dev-server-settings.json should not succeed` seems like it returned the file even though the result is the no route found page

The request should be in the format:
`http://localhost:8888/VAADIN/@fs/C:/Users/mikae/Code/flow/flow-tests/test-frontend/vite-basics/target/vaadin-dev-server-settings.json` to work correctly and get 403 so get fsUrl should take into account windows file path and use `FrontendUtils.getUnixPath` for the file path and check to add a `/` if not exists.

**Review by caalador — CHANGES_REQUESTED** [review]

File: flow-tests/test-frontend/vite-basics/src/test/java/com/vaadin/viteapp/FileAccessIT.java (hunk 1 above)
Comment (Artur-) [reply flow-tests/test-frontend/vite-basics/src/test/java/com/vaadin/viteapp/FileAccessIT.java]: Does it work on windows now?

File: flow-tests/test-frontend/vite-basics/src/test/java/com/vaadin/viteapp/FileAccessIT.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                    "Request for " + fileInProject + " should have failed",
+                    e.getMessage().contains(
+                            "Server returned HTTP response code: 403"));
+        }
+
+    }
+
+    private URL getFsUrl(String fileInProject) throws IOException {
+        // For Windows, the URLs should be like
+        // http://localhost:8888/VAADIN/@fs/C:/Users/mikae/Code/flow/flow-tests/test-frontend/vite-basics/target/vaadin-dev-server-settings.json
```
Comment (caalador) [inline flow-tests/test-frontend/vite-basics/src/test/java/com/vaadin/viteapp/FileAccessIT.java:77]:
```suggestion
        // http://localhost:8888/VAADIN/@fs/C:/Code/flow/flow-tests/test-frontend/vite-basics/target/vaadin-dev-server-settings.json
```

TMI I don't need my paths there. else works now with windows

**Review by caalador — APPROVED** [review]

**Pull request merged**